### PR TITLE
Improve CMS customization

### PR DESCRIPTION
**Pull request by tiagoalvesdulce**
Now that I played with this a little, I'm opening this WIP so we can discuss some points and open it to collaboration.

* Politeia's presentational components use a lot of props with the word proposal eg. `fetchProposal`, `proposals`, `proposalCount`, etc. passed by connectors. IMO, to reduce the number of if/else inside these components we should rename all these variables to "registers" or "data". This way we could create custom connectors easily without touching presentational components. What do you think?

### CODE REVIEW

**Comment by alexlyp** [comment]
Changes look great. 

I think the few isCMS that have been added aren't too overbearing.  In the end, it's really your guys' call. Mostly, I hope to not interfere too much with ongoing politeiagui dev.  Obviously, with the upcoming snew removal/replacement this should be even easier to keep seperated.

**Comment by alexlyp** [comment]
Figured I'd leave this here so we can keep track of various forms/pages that will need to be implemented for the CMS MVP:

- [ ] Admin Invite
- [ ] Contractor Registration
- [ ] Submit Invoice
- [ ] User View Invoices
- [ ] Admin View Invoices (by status and/or month)
- [ ] Admin Review Invoices with commenting and approving/updating status
- [ ] Basic "Pay invoice" by Admin
- [ ] User Invoice Overview

**Comment by thi4go** [comment]
Hey guys, just updating the MVP checklist:

- [x] Admin Invite
- [x] Contractor Registration
- [x] Submit Invoice
This will basically be 2 dropdowns for month and year. Then a text area to input the csv/line item information. In the future we can do some fancier input, like an Add line item form.
- [x] User View Invoices
This should display a list of invoices for a logged in user. Display month/year/total amount/status
- [ ] Invoice Details
Upon clicking on an invoice in the list above, it should show the raw csv/line items and the various details. If an Admin has commented on the invoice, show those as well.
- [ ] Admin View Invoices (by status and/or month)
Same list as above, but displayed by status and invoiced month
- [ ] Admin Review Invoices with commenting and approving/updating status
SImiliar to Invoice Details but allows for administrators to comment and also to approve/update status.
- [ ] Basic "Pay invoice" by Admin
For MVP we should have a basic "pay" invoices functionality for admins. This should at least allow the admin to enter a txid of the payment transaction. More to come on this in the future.
- [ ] User Invoice Overview
Landing page for Users. Maybe showing total amounts, total invoices per status, etc.

**Comment by fernandoabolafio** [comment]
I fixed the errors in the invite user page and also the failing tests for Travis. 
After dealing with the invoice submission errors, what do you think if we merge this PR, move the checklist to an issue and open new PRs regarding the missing feature?

**Comment by fernandoabolafio** [comment]
Okay guys, so let's make this PR integrate the CMS functionalities which are already on master and/or introduced on https://github.com/decred/politeia/pull/756. Which means this PR should cover:
 - Admin Invite
 - Contractor Registration
 - Submit Invoice
 - User View Invoices
 - Invoice Details
 - Admin View Invoices (by status and/or month)

The remaining functionalities we can open separated PRs to address them.

**Comment by thi4go** [comment]
Hey guys, I pushed the invoice detail page. Travis is mad at me again, i'll deal with him first thing tomorrow. Planning to work on the edit invoice feature in the morning.

File: src/components/ProposalDetail/ProposalDetail.js [hunk 1]
```diff
@@ -140,10 +140,21 @@ class ProposalDetail extends React.Component {
...
       tempThreadTree,
+      isCMS,
       ...props
     } = this.props;
     const comments = this.state.sortedComments;
     const tempTree = tempThreadTree[commentid];
+    const data = !isCMS
+      ? proposalToT3(proposal, 0).data
+      : invoiceToT3(proposal, 0).data;
+    const selftext = !isCMS
```
Comment (tiagoalvesdulce) [inline src/components/ProposalDetail/ProposalDetail.js]: instead of combining multiple ternary operators, go with `switch` or `if` to improve readability.

**Pull request merged**